pkg/artifactory/resource_artifactory_local_repository.go

```diff
@@ -268,6 +275,11 @@ func resourceLocalRepositoryExists(d *schema.ResourceData, m interface{}) (bool,
 
 	_, resp, err := c.V1.Repositories.GetLocal(context.Background(), d.Id())
 
+	if resp != nil {
```

Comment:
The equality check should be `resp == nil` but check `err` first. If you check error first you probably never need to check if `resp` is `nil`.